src/test/java/me/zilzu/mycoupon/application/service/CouponApplyTest.java

```diff
@@ -51,10 +57,50 @@ void test4() {
 
         Coupon coupon = couponService.create(couponCreationRequest);
         Coupon foundCoupon = couponService.retrieve(coupon.id);
-        couponService.apply(foundCoupon.id);  // valid false
+        couponService.apply(foundCoupon.id, 1000d);  // valid false
 
         assertThatThrownBy(() -> {
-            couponService.apply(coupon.id);
+            couponService.apply(coupon.id, 1000d);
         }).isInstanceOf(RuntimeException.class);
     }
+
+    @Test
+    @DisplayName("쿠폰 history 에 할인된 가격이 표시된다.DiscountType AMOUNT 일 때")
+    void test5() {
+        CouponCreationRequest couponCreationRequest = new CouponCreationRequest(CouponDuration.ONCE, null, DiscountType.AMOUNT, 10000L, null);
+
+        Coupon coupon = couponService.create(couponCreationRequest);
+        Coupon foundCoupon = couponService.retrieve(coupon.id);
+
+        double price = 1000d;
+        couponService.apply(foundCoupon.id, price);
+
+        List<CouponHistory> couponHistories = couponHistoryService.retrieveCouponHistoryList(foundCoupon.id);
+
+        for (CouponHistory couponHistory : couponHistories) {
+            CouponHistory foundCouponHistory = couponHistoryService.retrieveCouponHistory(new CouponHistoryId(couponHistory.id));
+            assertThat(foundCouponHistory.price).isEqualTo(price);
+            assertThat(foundCouponHistory.discountedPrice).isEqualTo(price - foundCoupon.amountOff);
+        }
+    }
+
+    @Test
+    @DisplayName("쿠폰 history 에 할인된 가격이 표시된다. DiscountType PERCENTAGE 일 때")
+    void test6() {
+        CouponCreationRequest couponCreationRequest = new CouponCreationRequest(CouponDuration.ONCE, null, DiscountType.PERCENTAGE, null, 10d);
+
+        Coupon coupon = couponService.create(couponCreationRequest);
+        Coupon foundCoupon = couponService.retrieve(coupon.id);
+
+        double price = 10d;
+        couponService.apply(foundCoupon.id, price);
+
+        List<CouponHistory> couponHistories = couponHistoryService.retrieveCouponHistoryList(foundCoupon.id);
+
+        for (CouponHistory couponHistory : couponHistories) {
+            CouponHistory foundCouponHistory = couponHistoryService.retrieveCouponHistory(new CouponHistoryId(couponHistory.id));
+            assertThat(foundCouponHistory.price).isEqualTo(price);
+            assertThat(foundCouponHistory.discountedPrice).isEqualTo(price * foundCoupon.percentOff / 100);
```

Comment:
음.. 보통 assert 문에서는 이런 로직보다는 실제 값을 넣는게 더 합당해보임.

예를 들어, price 가 10이고, discount percentOff 가 10 이라면,
10에서 10% 할인이니까, 9가 되는게 맞음.

```suggestion
            assertThat(foundCouponHistory.discountedPrice).isEqualTo(9);
```

처럼 작성하는게 일반적일듯.

+ 이렇게 하면 위에서 언급한 것처럼 Double 연산 소수점 이슈로 assert문 통과안할 수 있음. (예를 들어, price: 10.5, percentOff: 10.5 과 같은 경우)

Reply:
아하... 연산 그대로하면 테스트가 있으나 마나한 의미겠네여